claro.theme

```diff
@@ -179,3 +180,79 @@ function claro_form_media_form_alter(&$form, FormStateInterface $form_state) {
     $form['#attached']['library'][] = 'claro/media-form';
   }
 }
+
+/**
+ * Implements hook_preprocess_form_element().
+ */
+function claro_preprocess_form_element(&$variables) {
+  if (!empty($variables['element']['#errors'])) {
+    $variables['label']['#attributes']['class'][] = 'has-error';
+  }
+
+  if ($variables['disabled']) {
+    $variables['label']['#attributes']['class'][] = 'is-disabled';
+
+    if (!empty($variables['description']['attributes'])) {
+      $variables['description']['attributes']->addClass('is-disabled');
+    }
+  }
+}
+
+/**
+ * Implements template_preprocess_HOOK() for input.
+ */
+function claro_preprocess_input(&$variables) {
+  $type_api = $variables['element']['#type'];
+  $type_html = $variables['attributes']['type'];
+  $text_types_html = [
+    'text',
+    'email',
+    'tel',
+    'number',
+    'search',
+    'password',
+    'date',
+    'time',
+    'file',
+    'color',
+    'datetime-local',
+    'url',
+    'month',
+    'week',
+  ];
+
+  $variables['attributes']['class'][] = 'form-element';
+  $variables['attributes']['class'][] = Html::getClass('form-element--type-' . $type_html);
+  $variables['attributes']['class'][] = Html::getClass('form-element--api-' . $type_api);
+
+  if (in_array($type_html, $text_types_html)) {
+    $variables['attributes']['class'][] = 'form-element--basic';
```

Comment:
I would suggest that we only add form-element classes for the text type inputs. Other inputs should be considered their own components and separate from this.

Reply:
okay